core/editor.js

```diff
@@ -492,9 +492,10 @@
 
 	function loadPlugins( editor ) {
 		var config = editor.config,
-			plugins = config.plugins,
-			extraPlugins = config.extraPlugins,
-			removePlugins = config.removePlugins;
+			// We have to trim plugins so they will be correctly processed by our regex.
+			plugins = CKEDITOR.tools.trim( config.plugins ),
+			extraPlugins = CKEDITOR.tools.trim( config.extraPlugins ),
+			removePlugins = CKEDITOR.tools.trim( config.removePlugins );
```

Comment:
Wouldn't it be better to replace all whitespace here and then simplify `split` and `replace` statements as there won't be any whitespace to deal with? Because now we have separate treating for spaces at the beginning and end of the strings and separate for spaces inside strings.